### PR TITLE
Create find_root_vertex.py

### PR DESCRIPTION
**Pull request by amamisha59**
### Describe your change:

added the algorithm for finding the root vertex using python programming

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: graphs/find_root_vertex.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    def __init__(self, edges, n):
+        # Resize the list to hold `n` elements
+        self.adj = [[] for _ in range(n)]
+
+        # Add an edge from source to destination
+        for edge in edges:
+            self.adj[edge[0]].append(edge[1])
+
+# Function to perform DFS traversal on the graph
+def DFS(graph, v, discovered):
```
Comment (cclauss) [inline graphs/find_root_vertex.py:17]:
```suggestion
def depth_first_search(graph, v, discovered):
```

File: graphs/find_root_vertex.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    discovered[v] = True  # Mark the current node as discovered
+
+    # Do for every edge (v, u)
+    for u in graph.adj[v]:
+        if not discovered[u]:  # `u` is not discovered
+            DFS(graph, u, discovered)
+
+# Function to find the root vertex of a graph
+def findRootVertex(graph, n):
```
Comment (cclauss) [inline graphs/find_root_vertex.py:26]:
```suggestion
def find_root_vertex(graph, n):
```

File: graphs/find_root_vertex.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    # Build a directed graph from the given edges
+    graph = Graph(edges, n)
+
+    # Find the root vertex in the graph
+    root = findRootVertex(graph, n)
+
+    if root != -1:
+        print('The root vertex is', root)
+    else:
+        print('The root vertex does not exist
```
Comment (tianyizheng02) [inline graphs/find_root_vertex.py:67]: You missed the last few characters of this line when you were copy-pasting this code from the web. Really gave it away that you didn't write this yourself.

**Review by tianyizheng02 — COMMENTED** [review]
I found the exact same code on [this webpage](https://www.techiedelight.com/root-vertex-graph/). @cclauss, if you check the original commit for this PR, you'll see that the code matches down to the comments. IDK if this site is where the contributor found this implementation, but this confirms that it was plagiarized from _somewhere_ on the web.

**Pull request closed without merging**